libsql-server/src/http/user/listen.rs

```diff
@@ -6,18 +6,30 @@ use crate::{
     namespace::{NamespaceName, NamespaceStore},
 };
 use axum::extract::State as AxumState;
-use axum::http::header::{CACHE_CONTROL, CONTENT_TYPE};
-use axum::http::{HeaderValue, Uri};
-use axum::response::{IntoResponse, Redirect, Response};
-use axum_extra::{extract::Query, json_lines::JsonLines};
+use axum::http::Uri;
+use axum::response::{
+    sse::{Event, Sse},
+    IntoResponse, Redirect,
+};
+use axum_extra::extract::Query;
 use futures::{Stream, StreamExt};
 use hyper::HeaderMap;
 use serde::{Deserialize, Serialize};
+use std::boxed::Box;
+use std::convert::Infallible;
+use std::pin::Pin;
+use std::time::Duration;
 use tokio_stream::wrappers::errors::BroadcastStreamRecvError;
 
 use super::db_factory::namespace_from_headers;
 use super::AppState;
 
+const LAGGED_MSG: &str = "some changes were lost";
```

Comment:
I wonder if there is a way we can encode this error better so a user doesn't have to string match on a lagged event vs other types of errors.

Reply:
I'm guessing we could do something like this:

```rs
#[derive(Debug, Serialize)]
#[serde(tag = "code", content = "message")]
enum ErrorEvent {
    Lagged(String),
    Other(String),
}
```

and then use that. But, this part of the code I didn't change, I just moved the position of it.